socket_SUITE: do not fail on IPv6-only nodes

max-au opened the pull request:
Before this change, socket_SUITE fails most of IPv4 tests
with IPv4 address available only on loopback adapter. With this
patch:
 - IPv4-specific tests will be skipped on IPv6-only nodes
 - non-specific tests will utilise either v4 or v6, depending
   on availability (with v4 preferred)
 - minor cleanup (removed unused suite/doc functions)

bmk commented:
It looks ok, but there are now merge conflicts. Could you please rebase on the lastest master.

max-au commented:
Rebased to master.

One more note, I removed the commend stating that `peer` is not able to run nodes on a different host. This isn't true, `peer` is supports starting nodes that way, using `host` parameter. There is an example of doing that over SSH and even an example with docker container (which is technically separate host).

bmk commented:
Yes, I know that but the changes where not made such (the host value was just ignored).
